### ISSUE TITLE
Please add sftp.extra_args

### ISSUE THREAD

**Issue by yacoob** [restic/restic#4241]
Output of `restic version`
--------------------------
restic 0.15.1 compiled with go1.19.5 on linux/amd64

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------
I need to point ssh restic is using to a specific keyfile. In order to provide both repository and ssh command like this:

```
restic  \
  --option sftp.command=ssh\ -p\ 23\ -l\ abcd\ -i\ /etc/restic/keys/foo\ targethost\ -s\ sftp   \
  --repo sftp://abcd@targethost:23
```

Can you please add something like `sftp.extra_args` that would be used directly in `buildSSHCommand` and just added to the command line? I can then say `--option sftp.command=-i\ /etc/restic/keys/foo` and the whole thing would look cleaner.

The solution suggested in the manual (using `~/.ssh/config`) is nice, but I'd really like to keep all config file in a single place, and avoid having them intermixed with otherwise unrelated entries in ssh's config. I can make a separate config file for this purpose, but then I need to point ssh to it, which brings us to the same problem as my initial one :)

What are you trying to do? What problem would this solve?
---------------------------------------------------------
The `restic` command line invocation would look cleaner.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Eyeballing `restic` as borgbackup replacement; doing so would allow me to use a cheaper backend storage :)

Thanks!

**Comment by nettnikl** [comment]
Hi, having the same problem. Workaround is presented in https://github.com/restic/restic/issues/312#issuecomment-348924867 and https://github.com/restic/restic/issues/312#issuecomment-482903282.

Seeing https://github.com/restic/restic/blob/1733b24f9d462407e8fe9893332e2f1cc093076a/src/restic/backend/sftp/sftp.go#L140 however, i don't think it's to much of an issue to add this extra option, and would be much, much cleaner.

**Comment by quite** [comment]
Yes, an option to influence how buildSSHCommand works is much needed. Precisely for adding custom options such as `-i file` to the resulting ssh commandline, without having to write the whole `ssh ...` commandline oneself. Having to do that also makes the host:port part of the sftp--repo URI useless.

**Comment by adamantike** [comment]
I have created #4519, looking forward to solve this issue and make those workarounds obsolete.

**Comment by adamantike** [comment]
`sftp.args` is now available in version `0.16.1`.
